### PR TITLE
feat: web wallet composite

### PR DESCRIPTION
**Pull request by LeoCourbassier**
## Context
By using the hooks in the React package, we were able to create a WebWallet and ship it in the @fuels/react package.
This can be enabled by using `showWebWallet` in the `FuelProvider`.
The default is disabled.

## This PR
- Adds a new component to improve UX
- Depends on https://github.com/FuelLabs/fuel-explorer/pull/467

## Screenshots
In the comments below.

### CODE REVIEW

File: packages/react/src/providers/FuelProvider.tsx [hunk 1]
```diff
@@ -24,6 +25,7 @@ export function FuelProvider({
       <FuelHooksProvider fuelConfig={fuelConfig}>
         <FuelUIProvider theme={theme} fuelConfig={fuelConfig}>
           <Connect />
+          <WebWallet />
```
Comment (luizstacio) [inline packages/react/src/providers/FuelProvider.tsx:28]: I think this should be optional, added by the DApp in a opt-in manner.

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: packages/react/src/providers/FuelProvider.tsx (hunk 1 above)
Comment (LeoCourbassier) [reply packages/react/src/providers/FuelProvider.tsx]: It makes total sense. I have updated the PR

**Comment by LeoCourbassier** [comment]
Depends on https://github.com/FuelLabs/fuel-explorer/pull/569

**Comment by LeoCourbassier** [comment]
### Light Theme

**Comment by LeoCourbassier** [comment]
### Dark Theme

File: packages/react/src/providers/FuelProvider.tsx [hunk 2]
```diff
@@ -10,6 +11,7 @@ export { useConnectUI } from './FuelUIProvider';
 
 type FuelProviderProps = {
   ui?: boolean;
+  showWebWallet?: boolean;
```
Comment (LuizAsFight) [inline packages/react/src/providers/FuelProvider.tsx:14]: can we invert it to `hideWebWallet` ? then the wallet will be available by default

**Review by LuizAsFight — CHANGES_REQUESTED** [review]

File: examples/react-app/package.json [hunk 3]
```diff
@@ -18,12 +18,12 @@
     "@wagmi/core": "2.13.4",
     "clsx": "2.1.1",
     "fuels": "0.94.6",
-    "react": "18.3.1",
-    "react-dom": "18.3.1"
+    "react": "18.2.0",
+    "react-dom": "18.2.0"
```
Comment (LuizAsFight) [inline examples/react-app/package.json:22]: why we downgrading react deps ?

File: examples/react-app/package.json [hunk 4]
```diff
@@ -18,12 +18,12 @@
...
   },
   "devDependencies": {
     "@tanstack/react-query-devtools": "5.35.1",
-    "@types/react": "18.3.1",
+    "@types/react": "18.2.54",
```
Comment (LuizAsFight) [inline examples/react-app/package.json:26]: why we downgrading react deps ?

File: examples/react-next/package.json [hunk 5]
```diff
@@ -21,7 +21,7 @@
   "devDependencies": {
     "@tanstack/react-query-devtools": "5.35.1",
     "@types/node": "20.12.11",
-    "@types/react": "18.3.1",
+    "@types/react": "18.2.54",
```
Comment (LuizAsFight) [inline examples/react-next/package.json:24]: why we downgrading react deps ?

**Review by LuizAsFight — COMMENTED** [review]
check all packages.json to see if we're unnecessarily downgrading any deps

File: packages/react/src/ui/WebWallet/hooks/useBalanceFormat.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        precision: value.isZero() ? 1 : DEFAULT_MIN_PRECISION,
+        units: decimals,
+      }),
+      formattedBalanceFull: format(value, {
+        ...formatOpts,
+        units: decimals,
+      }),
+    }),
+    [value, decimals],
+  );
```
Comment (LuizAsFight) [inline packages/react/src/ui/WebWallet/hooks/useBalanceFormat.ts:29]: we need to review if have the same behavior of next fuel wallet now after [this PR](https://github.com/FuelLabs/fuels-wallet/pull/1492) got merged

File: packages/react/src/ui/WebWallet/index.css [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+    @media (min-width: 1025px) {
+        & > .fuel-PopoverContent {
+            height: 50vh;
+            max-height: 50vh;
+            width: 370px;
+            max-width: 370px;
+        }
+    }
+}
```
Comment (LuizAsFight) [inline packages/react/src/ui/WebWallet/index.css:29]: we should use the breakpoints to be coherent with other components

**Review by LuizAsFight — CHANGES_REQUESTED** [review]

**Comment by LeoCourbassier** [comment]
This feature will continue on https://github.com/FuelLabs/fuel-connectors/pull/281

**Pull request closed without merging**